File: arrays/containerWithMostWater.py
def maxArea(height):
    left = 0
    right = len(height) - 1
    max = 0

    while left < right:
        minVal = min(height[left], height[right])
        area = minVal * (right - left)
        if area > max:
            max = area
        
        if minVal == height[left]:
            left += 1
        else:
            right -= 1
    
    return max

File: arrays/containerWithMostWater/containerWithMostWater.js
const maxArea = (height) => {
    let highArea = 0;
    let left = 0;
    let right = height.length - 1;

    while (left < right) {
        let rightVal = height[right];
        let leftVal = height[left];
        let indDif = right - left;
        let area = Math.min(rightVal, leftVal) * indDif;
        if (area > highArea) highArea = area;

        if (leftVal < rightVal) left++;
        else right--;
    }

    return highArea
};